cloud/vmware/vsphere_guest.py

```diff
@@ -545,9 +551,14 @@ def deploy_template(vsphere_client, guest, resource_pool, template_src, esxi, mo
     try:
         if vmTarget:
             changed = False
+        elif snapshot_to_clone != None:
```

Comment:
When testing whether a variable is None idiomatic python uses ```if snapshot_to_clone is None```.  The reason is that None is a singleton (https://en.wikipedia.org/wiki/Singleton_pattern ) so we use a test to see if they are the same object instead of a test that the values are equal.